lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -433,17 +527,31 @@ class BuildOpGroups : public calyx::FuncOpPartialLoweringPattern {
 
     /// Create assignments to the inputs of the library op.
     auto group = createGroupForOp<TGroupOp>(rewriter, op);
+
+    bool isSeqCondCheck = false;
+    calyx::RegisterOp condReg = nullptr, resReg = nullptr;
+    if (isa<CmpIOp>(op)) {
+      auto cmpIOp = cast<CmpIOp>(op);
```

Comment:
```suggestion
    if (auto cmpIOp = dyn_cast<CmpIOp>(op)) {
```

Reply:
This one requires static check so `dyn_cast` didn't work when I tried to compile it.
Other than that, I have fixed everything based on your review!